src/WebMapTextureLoader.cpp

```diff
@@ -1,75 +1,128 @@
 
-namespace csp::simpleWmsBodies {
+#include "utils.hpp"
+
+#include "../../../src/cs-utils/filesystem.hpp"
+#include "../../../src/cs-utils/logger.hpp"
+
+#include <curlpp/Infos.hpp>
+#include <curlpp/Options.hpp>
+
+#define STB_IMAGE_IMPLEMENTATION
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <stb_image.h>
+#include <stb_image_write.h>
+
+namespace csp::simplewmsbodies {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
 
 WebMapTextureLoader::WebMapTextureLoader()
-: mThreadPool(32) {
- }
+    : mThreadPool(32) {
+}
 
-WebMapTextureLoader::~WebMapTextureLoader() {
+////////////////////////////////////////////////////////////////////////////////////////////////////
 
+WebMapTextureLoader::~WebMapTextureLoader() {
 }
 
-bool fileExist(const char *fileName)
-{
-    std::ifstream infile(fileName);
-    return infile.good();
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+bool fileExist(const char* fileName) {
+  std::ifstream infile(fileName);
+  return infile.good();
 }
 
-std::string WebMapTextureLoader::loadTexture(std::string time, std::string requestStr, std::string name, std::string format) {
-    std::string cacheDir = "../share/resources/textures/" + name + "/";
-    std::string year;
-    std::stringstream time_stringstream(time);
-    std::getline(time_stringstream,year, '-');
-    cacheDir+= year + "/";
-    auto cacheDirPath(boost::filesystem::absolute(boost::filesystem::path(cacheDir)));
-    if (!(boost::filesystem::exists(cacheDirPath))) {
-      try {
-        cs::utils::filesystem::createDirectoryRecursively(
-            cacheDirPath, boost::filesystem::perms::all_all);
-      } catch (std::exception& e) {
-        std::cerr << "Failed to create cache directory: " << e.what() << std::endl;
-      }
-    }
-    requestStr += "&TIME=";
-    
-    requestStr += time;
-    std::replace(time.begin(),time.end(),'/','-');
-    std::string cacheFile = cacheDir + time + ".png";
-    if(fileExist(cacheFile.c_str())) {
-        return cacheFile;
-    }
-    std::ofstream out;
-    out.open(cacheFile, std::ofstream::out | std::ofstream::binary);
-    if (!out) {
-        std::cerr << "Failed to open for writing!" << std::endl;
-    }
-    curlpp::Easy request;
-    request.setOpt(curlpp::options::Url(requestStr));
-    request.setOpt(curlpp::options::WriteStream(&out));
-    request.setOpt(curlpp::options::NoSignal(true));
-    request.perform();
+////////////////////////////////////////////////////////////////////////////////////////////////////
 
-    out.close();
-    if(curlpp::infos::ResponseCode::get(request) == 400) {
-        remove(cacheFile.c_str());
-        return "Error";
+std::string WebMapTextureLoader::loadTexture(
+    std::string time, std::string requestStr, std::string const& name, std::string const& format) {
+  std::string       cacheDir = "../share/resources/textures/" + name + "/";
```

Comment:
In my opinion, the cache directory should behave similar to the cache directory of `csp-lod-bodies`. That means it should be configurable via a `map-cache` key in the settings. The default value could be `"map-cache"` which would be a direcory in the `bin` directory, next to the `cosmoscout` executable.